### DISCUSSION TITLE
Warned by Bluelink Canada to stop unusual activities

### DISCUSSION THREAD

**Discussion by CrAsH0v3r** [Hacksore/bluelinky#225] [General]
Just to share, I've been warned by Hyundai Canada that there was "unusual activites" on my account and that they would suspend my account if I don't call them back. They called multiple times without telling the reason and they finally gave the real reason by email. 

I stopped my bluelinky process that was running 4 times a day and replyed to the mail saying I took appropriate action, and my account is still working 4 days past the deadline.

So no more Bluelinky for me.. at least for now :(

**Comment by Hacksore** [comment]
@CrAsH0v3r Thanks for starting this discussion. It sucks they want to stop you from having control of your vehicle 😡.

I guess for them it's pretty easy to check their logs and see what [user agent](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent) a client is hitting their API since we don't set one for Canada.

https://github.com/Hacksore/bluelinky/blob/master/src/controllers/canadian.controller.ts#L119

The default user agent for the request lib we use (got) has a default value making it easy for Hyundai to track and consider you "suspicious".

https://github.com/sindresorhus/got/blob/main/source/core/options.ts#L720

We can chose to add some obfuscation like using the same user agent that the mobile app uses to try and "blend in" but i'm sure they might have other fingerprinting tactics that could also make you get flagged.

**Comment by aronsysdone** [comment]
> We can chose to add some obfuscation like using the same user agent that the mobile app uses to try and "blend in" but i'm sure they might have other fingerprinting tactics that could also make you get flagged.

I don't think they have a sophisticated flagging system. Obfuscating the UserAgant goes a long way.
Need to do it.
